src/HotChocolate/Core/test/Types.Tests/Types/Relay/NodeTypeTests.cs

```diff
@@ -243,6 +243,43 @@ async Task Error() => await new ServiceCollection()
         error.Message.MatchSnapshot();
     }
 
+    [Fact]
+    public async Task Infer_Node_From_Query_As_Interface_From_Interface()
+    {
+        await new ServiceCollection()
+            .AddGraphQL()
+            .AddQueryType<Query10>()
+            .AddType<Query10.DrivingLicense>() // TODO: Should this be discovered?
+            .AddGlobalObjectIdentification()
+            .BuildSchemaAsync()
+            .MatchSnapshotAsync();
+    }
+
+    [Fact]
+    public async Task Infer_Node_From_Query_As_Interface_From_Abstract()
+    {
+        await new ServiceCollection()
+            .AddGraphQL()
+            .AddQueryType<Query11>()
+            .AddType<Query11.DrivingLicense>() // TODO: Should this be discovered?
```

Comment:
// TODO: Should this be discovered?

Reply:
https://github.com/ChilliCream/graphql-platform/pull/6103#discussion_r1189473947